jsoniter-scala-benchmark/shared/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/benchmark/NinnyFormats.scala

```diff
@@ -25,14 +41,17 @@ implicit val jsoniterWritingCodec = new JsonValueCodec[JsonValue] {
             out.writeObjectEnd()
         case JsonDecimal(preciseValue) => 
           if(preciseValue.precision == 0)
-          out.writeVal(preciseValue.toBigInt)
+            out.writeVal(preciseValue.toBigInt)
           else
-          out.writeVal(preciseValue)
+            out.writeVal(preciseValue)
         case JsonDouble(value) => 
-          if(value % 1 == 0)
-          out.writeVal(BigDecimal(value).toBigInt)
+          val longValue = value.toLong
+          if(longValue == value)
+            out.writeVal(longValue)
+          else if(value % 1 == 0)
```

Comment:
This branch is redundant because `Long` contains all values of `Double` rounded to a whole number.

Reply:
Is that right? I long ago forgot the details of number representations, but some primitive experimentation indicates that double can be much larger https://scastie.scala-lang.org/jrSbT5FgT2OWgOWRuoH9RQ

Reply:
I mean that the mantissa of `Double` can be stored in `Long` for whole numbers without loosing any bit of information.

Are you going to print all those 300 zeroes? Do you have any reason for that overhead?

Reply:
Ah, yes you're right I'm that regard.
No, I wouldn't print the zeros, just use the scientific notation. 

I think I'll keep that branch, although you are right that it's a little silly. The user should probably have been using the JsonDecimal in that case.

Reply:
I recommend just to remove that `else if(value % 1 == 0)` branch. 

You can check how much overhead it adds using the `ArrayOfDoublesWriting` benchmark.

Reply:
Much faster, those numbers still have a trailing `.0` but that's not such a big deal since they're so large. 
There are a few places where it seems like jsoniter doesn't write numbers as compactly as it could (removing trailing `.0` and scientific notation if there are more than a couple`0`s in the least significant digits). Would you mind if I took a stab at a PR for some of those? If not, any tips?

Reply:
The jsoniter-scala library use Java's formatting for `Double` and `Float` numbers.
We can add a new configuration option for the writer to serialize them into as short as possible sequence of bytes and implement it.